main.js

```diff
@@ -44,34 +55,84 @@ function draw() {
     points.push([x, y]);
   }
 
-  const minX = Math.min(...points.map((p) => p[0]));
```

Comment:
The `...` method causes a stack overflow when there are too many points. This is replaced with a regular for-loop